### ISSUE TITLE
wasFatal method from Log::Report::Dispatcher::Try changed

### ISSUE THREAD

**Issue by MadLord80** [markov2/perl5-XML-Compile-SOAP#5]
Hi!
Since 1.28 version of Log::Report method wasFatal() from Log::Report::Dispatcher::Try changed. Now it return boolean ($_->isFatal) instead exception ($self->{exceptions}[-1]).
But in XML::Compile::SOAP::Server (version 3.26) code for compileHandler() method not changed.
```
if($decode)
        {   $data = try { $decode->($xmlin) };
            if($@)
            {   $@->wasFatal->throw(reason => 'INFO', is_fatal => 0);
                return ( RC_NOT_ACCEPTABLE, 'input validation failed'
                   , $self->faultValidationFailed($name, $@->wasFatal))
            }
        }
```
After `$@->wasFatal->throw` - `$@->wasFatal` is not existing. And calling `$exception->message` in faultValidationFailed() method in XML::Compile::SOAP11::Server returns an error (call on undefined).

**Comment by markov2** [comment]
IMO, `wasFatal()` still returns the exception, but something has changed such that `$@` is overwritten.  Function `first()` works like `grep()` but returns only one of the filtered list.  It is not like `map()`.  You are probably fooled by the overloading of the exception object.

My own (current) preferred way to avoid that is
```perl
if(my $e = $@->wasFatal)
```
Catch the exception immediately.  It is also possible is
```perl
if(my $err = $@)
```
to catch the error object before someone breaks it.

So: you may have encountered a real bug in ::SOAP.  I will make above change... can you confirm that it works?  Small changes on 3 lines into:
```perl
            if(my $e = $@->wasFatal)
            {   $e->throw(reason => 'INFO', is_fatal => 0);
                return ( RC_NOT_ACCEPTABLE, 'input validation failed'
                   , $self->faultValidationFailed($name, $e))
            }
```

**Comment by MadLord80** [comment]
> Function first() works like grep() but returns only one of the filtered list.

Ok,  i understand.

>  you may have encountered a real bug in ::SOAP

Yes, that's what I meant. Bug in SOAP, not in Report.

> can you confirm that it works?

Yes, it works!

**Comment by markov2** [comment]
There haven't been any bugs reported in 2 years, so probably it will be quite for some months to go.  Therefore, I have released 3.27 for you.

**Comment by MadLord80** [comment]
Thanks!
